test/gtest/sequences.cpp

```diff
@@ -0,0 +1,207 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <miopen/sequences.hpp>
+#include <gtest/gtest.h>
+
+namespace Seq = miopen::seq;
+
+TEST(CPU_sequence_NONE, SimpleListTest)
+{
+    using TestSequence = Seq::Sequence<int, 1, 2, 4, 5>;
```

Comment:
Future improvements (nothing mentioned here needs to hold up this PR):

The documentation in sequences.hpp is quite confused. It explicitly states that values must be unique, but then gives a long example of a problem that only exists with duplicate values. The former and latter may apply to different subsets of the types defined here, but no effort is made to clarify.
...Actually, looking through the rest of that file, there are many more problems than just that...too much to talk about here.

Sequence must be tested with negative values and out-of-numerical-order values. Sign and numerical order aren't important, so we need to distinguish the behavior from other possibilities such as "magnitude is greater", "return the next value greater than the current", etc.

Add tests for ValidateValues().

Add tests for find().

The current ctor is broken since it uses assert (does not check for duplicates in Release build).
Related to that, add a test that enforces throwing on duplicated values.

I've found more issues, but I need to get back to my tickets. We can discuss tomorrow.